### PR TITLE
[ORANGE-1186] Notification Cron Crashes

### PR DESCRIPTION
**Pull request by mountHouli**
# Related Tickets

- [ORANGE-1186](https://jira.amida-tech.com/browse/ORANGE-1186)

# Other Repos' PR(s) Intended to Work With This PR

None.

# How Things Worked (or Didn't) Before This PR

The orange API runs a cron job every minute to see if there are any medication or meditation reminders that need to be sent. If so, it calls the auth service to login and calls the notification service to send the notifications.

Before this PR, if either of those API calls would fail, it would throw an unhandled error and therefore crash orange api.

# How Things Work Now (And How to Test)

It catches the errors and logs what happened.

**Note 1: This is intentionally branched off of https://github.com/amida-tech/orange-api/pull/161 . To save time, test them together by testing only this branch.** 

**Note 2: When looking at the code, it will be a little easier if you look commit-by-commit.**

### To Test:

1. Make sure notification still work by following the testing steps in https://github.com/amida-tech/orange-api/pull/161

2. Follow those same steps, except turn your auth service off. When the cron runs, you should get a nice error message and no crash. Repeat for the notification service.

# Readiness

1. [x] This PR has full test coverage (If this PR fixes a bug, you _must_ add test cases respresentative of the bug).
2. [x] This PR passes all automated tests.
3. [x] This PR has no linting errors.
4. [x] This PR has no hardcoded UI strings or other obvious issues.
5. [N/A] This PR has been updated to include all changes from develop.
6. [N/A] This PR's changes to configuration files have been documented in all appropriate places (such as but not limited to README.md), if applicable.
7. [N/A] This PR's required changes outside of the scope of this repository have been documented in this pull request.

### CODE REVIEW

File: lib/models/user/pushNotifications.js [hunk 1]
```diff
@@ -1,36 +1,52 @@
...
+            json: true,
+            body: {
+              username: config.pushNotificationsServiceUserUsername,
+              password: config.pushNotificationsServiceUserPassword
+            }
+        };
+
+        request.post(authArgs, function (err, res) {
+            if (err) {
+                console.log(`${new Date()} Error: pushNotifications.js: Push notificatios cron job call to Auth Service failed. Aborting.`, err);
```
Comment (ElijahGeorge) [inline lib/models/user/pushNotifications.js]: Is using the winston logger an option here?

File: lib/models/user/pushNotifications.js [hunk 2]
```diff
@@ -1,36 +1,52 @@
...
+                    "Content-Type": "application/json",
+                    "Authorization": "Bearer " + token
+                },
+                json: true,
+                body: { pushData }
+            };
+
+            request.post(pushNotificationArgs, function (err) {
+                if (err) {
+                    console.log(`${new Date()} Error: pushNotifications.js: Push notifications cron job call to Notification Service failed. Aborting.`, err);
```
Comment (ElijahGeorge) [inline lib/models/user/pushNotifications.js]: ditto.

File: lib/models/user/pushNotifications.js (hunk 1 above)
Comment (mountHouli) [reply lib/models/user/pushNotifications.js]: Thanks for the feedback! The amida winston logger would be great for this, but it's not implemented in this repo yet. This repo only uses the express winston express middleware, which, AFAIK, isn't for this purpose. Before implementing, I did a quick check, and it looks like `console.log` is the standard for this repo still.

File: lib/models/user/pushNotifications.js (hunk 2 above)
Comment (mountHouli) [reply lib/models/user/pushNotifications.js]: ditto.

Comment (ElijahGeorge) [reply lib/models/user/pushNotifications.js]: Got it. Thanks!

**Pull request merged**